cvat/apps/reid/static/reid/js/enginePlugin.js

```diff
@@ -0,0 +1,152 @@
+                                    if (jobData.stderr) {
+                                        message += ` Error: ${jobData.stderr}`;
+                                    }
+                                    showMessage(message);
+                                }
+                            }
+                        },
+                        error: (errorData) => {
+                            const message = `Can not check ReID merge. Code: ${errorData.status}. Message: ${errorData.responseText || errorData.statusText}`;
+                            showMessage(message);
+                        },
+                    });
+                }
+
+                setTimeout(checkCallback, 1000);
+            },
+            error: (errorData) => {
+                const message = `Can not start ReID merge. Code: ${errorData.status}. Message: ${errorData.responseText || errorData.statusText}`;
+                showMessage(message);
+            },
+            complete: () => {
+                reidButton.prop('disabled', false);
+            },
+        });
+    }
+
+    function cancel(reidButton) {
+        reidButton.prop('disabled', true);
+        $.ajax({
+            url: `/reid/cancel/${window.cvat.job.id}`,
+            type: 'GET',
+            success: () => {
+                reidButton.removeClass('run').text('Run ReID Merge').prop('disabled', false);
+            },
+            error: (errorData) => {
+                const message = `Can not cancel ReID process. Code: ${errorData.status}. Message: ${errorData.responseText || errorData.statusText}`;
+                showMessage(message);
+            },
+            complete: () => {
+                reidButton.prop('disabled', false);
+            },
+        });
+    }
+
+    const buttonsUI = $('#engineMenuButtons');
+    const reidWindowId = 'reidSubmitWindow';
+    const reidTresholdValueId = 'reidTresholdValue';
+    const reidDistanceValueId = 'reidDistanceValue';
+    const reidCancelMergeId = 'reidCancelMerge';
+    const reidSubmitMergeId = 'reidSubmitMerge';
+
+    const reidButton = $('<button> Run ReID Merge </button>').on('click', () => {
+        $('#taskAnnotationMenu').addClass('hidden');
+        if (reidButton.hasClass('run')) {
+            $('#annotationMenu').addClass('hidden');
+            userConfirm('ReID process will be canceld. Are you sure?', () => cancel(reidButton));
+        } else {
+            $('#annotationMenu').addClass('hidden');
+            $(`#${reidWindowId}`).removeClass('hidden');
+        }
+    }).addClass('menuButton semiBold h2').prependTo(buttonsUI);
+
+    $(`
+        <div class="modal hidden" id="${reidWindowId}">
+            <div class="modal-content" style="width: 300px; height: 170px;">
+                <table>
+                    <tr>
+                        <td> <label class="regular h2"> Treshold: </label> </td>
```

Comment:
Need to add hints. Also specify that the radius in pixels.

Reply:
Done